js/bitopro.js

```diff
@@ -0,0 +1,1349 @@
+        //                 "feeSymbol":"twd",
+        //                 "bitoFee":"0",
+        //                 "total":"0",
+        //                 "seq":"BNBTWD8984230567",
+        //                 "updatedTimestamp":1644902118265
+        //             }
+        //         ]
+        //     }
+        //
+        return this.parseTrades (trades, market, since, limit);
+    }
+
+    parseTransactionState (state) {
+        const states = {
+            'COMPLETE': 'ok',
+            'INVALID': 'failed',
+            'PROCESSING': 'pending',
+            'WAIT_PROCESS': 'pending',
+            'FAILED': 'failed',
+            'EXPIRED': 'failed',
+            'CANCELLED': 'failed',
+            'EMAIL_VERIFICATION': 'pending',
+            'WAIT_CONFIRMATION': 'pending',
+        };
+        return this.safeString (states, state, state);
+    }
+
+    parseTransaction (transaction, currency = undefined) {
+        //
+        // fetchDeposits
+        //             {
+        //                 "serial":"20220214X766799",
+        //                 "timestamp":"1644833015053",
+        //                 "address":"bnb1xml62k5a9dcewgc542fha75fyxdcp0zv8eqfsh",
+        //                 "amount":"0.20000000",
+        //                 "fee":"0.00000000",
+        //                 "total":"0.20000000",
+        //                 "status":"COMPLETE",
+        //                 "txid":"A3CC4F6828CC752B9F3737F48B5826B9EC2857040CB5141D0CC955F7E53DB6D9",
+        //                 "message":"778553959",
+        //                 "protocol":"MAIN",
+        //                 "id":"2905906537"
+        //             }
+        //
+        // fetchWithdrawals || fetchWithdraw
+        //             {
+        //                 "serial":"20220215BW14069838",
+        //                 "timestamp":"1644907716044",
+        //                 "address":"TKrwMaZaGiAvtXCFT41xHuusNcs4LPWS7w",
+        //                 "amount":"8.00000000",
+        //                 "fee":"2.00000000",
+        //                 "total":"10.00000000",
+        //                 "status":"COMPLETE",
+        //                 "txid":"50bf250c71a582f40cf699fb58bab978437ea9bdf7259ff8072e669aab30c32b",
+        //                 "protocol":"TRX",
+        //                 "id":"9925310345"
+        //             }
+        //
+        // withdraw
+        //             {
+        //                 "serial":"20220215BW14069838",
+        //                 "currency":"USDT",
+        //                 "protocol":"TRX",
+        //                 "address":"TKrwMaZaGiAvtXCFT41xHuusNcs4LPWS7w",
+        //                 "amount":"8",
+        //                 "fee":"2",
+        //                 "total":"10"
+        //             }
+        //
+        let currencyId = this.safeString (transaction, 'coin');
+        if (currencyId === undefined) {
+            currencyId = this.safeString (currency, 'code');
```

Comment:
We don't need to do lines 1110-1112, please, remove them.
Please, pass the second `currency` argument to `safeCurrencyCode`:

```JavaScript
        const code = this.safeCurrencyCode (currencyId, currency);
```